### PR TITLE
issue 3108 first attempt splitting KeyStore.getFirst

### LINKED ISSUE #3108: split getFirst into two methods: getFirstOptional and getFirstRequired
Almost all calls of `KeyStore.getFirst` assume that there will be a key returned. Many times, but not always, we also test it right below:

```ts
const primaryKi = await KeyStore.getFirst(this.acctEmail);
Assert.abortAndRenderErrorIfKeyinfoEmpty(primaryKi);
```

This is error prone, many times we forget to check.

But in some situations, we actually expect that that the key may not be present. You can tell such situations in case it follows with `if(primaryKi) { ... }` or something similar.

Todo in this issue:
 - [x] rename `KeyStore.getFirst` to `KeyStore.getFirstOptional` and change return signature to `Promise<KeyInfo | undefined>`
 - [x] add method `KeyStore.getFirstRequired` that contains just the call to `KeyStore.getFirstOptional` and `Assert.abortAndRenderErrorIfKeyinfoEmpty(primaryKi);`
 - [x] in the whole codebase, replace `getFirstOptional` with `getFirstRequired` wherever it seems to make sense - that is practically the whole codebase except maybe setup screen and other places, if any, that actually don't expect key to be present. 
- [x] where you switched the call to `getFirstRequired`, if that line followed with `Assert.abortAndRenderErrorIfKeyinfoEmpty(primaryKi);`, this check can now be removed

### PR DESCRIPTION
**Pull request by tomholub**
close #3108

### CODE REVIEW

**Comment by tomholub** [comment]
@danishnavid008 please merge changes from master to here, and help me finish this PR. Thanks!

**Comment by danishnavid008** [comment]
@tomholub This PR is ready for review

File: extension/js/common/assert.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ export class Assert {
 
   public static abortAndRenderErrOnUnprotectedKey = async (acctEmail?: string, tabId?: string) => {
     if (acctEmail) {
-      const primaryKi = await KeyStore.getFirstOptional(acctEmail);
+      const primaryKi = await KeyStore.getFirstRequired(acctEmail);
```
Comment (tomholub) [inline extension/js/common/assert.ts]: Here you switched to required, but you can see below that we are testing `&& primaryKi &&` for truthiness. That means it was expected that this key may be optional in this particular method. With this in mind, you should also scrutinize other parts.

**Review by tomholub — COMMENTED** [review]
I see one error - please have a look. Have you evaluated all usages of `getFirstRequired` vs `getFirstOptional`? Are there no situations when you're unsure and would want to double check with me?

File: extension/js/common/assert.ts (hunk 1 above)
Comment (danishnavid008) [reply extension/js/common/assert.ts]: @tomholub Yes, I've evaluated all usages of `getFirstRequired` vs `getFirstOptional`. I had mistakenly used `getFirstRequired` but then I fixed it in the next commit.

Comment (danishnavid008) [reply extension/js/common/assert.ts]: Can you please check the latest commit? You'll see I fixed it.

File: extension/chrome/settings/modules/change_passphrase.ts [hunk 2]
```diff
@@ -74,6 +73,7 @@ View.run(class ChangePassPhraseView extends View {
       await Ui.modal.error('Pass phrase did not match, please try again.');
       $('#current_pass_phrase').val('').focus();
     }
+    Assert.abortAndRenderErrorIfKeyinfoEmpty(this.primaryKi);
```
Comment (tomholub) [inline extension/chrome/settings/modules/change_passphrase.ts:76]: This one may have been added by mistake

**Review by tomholub — COMMENTED** [review]
It looks good now - there is one remaining line that maybe was added by mistake.

Maybe earlier I was reviewing an older version - not sure. Thanks!

File: extension/chrome/settings/modules/change_passphrase.ts (hunk 2 above)
Comment (danishnavid008) [reply extension/chrome/settings/modules/change_passphrase.ts:76]: @tomholub Yeah, It was added by mistake while resolving the merge conflicts

**Comment by tomholub** [comment]
All good, thank you

**Pull request merged**